src/client/Microsoft.Identity.Client/MsalManagedIdentityException.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Identity.Client.ManagedIdentity;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// This exception class is for exceptions generated from Managed Identity sources.
+    /// For more details, see https://aka.ms/msal-net-exceptions
+    /// </summary>
+    public class MsalManagedIdentityException : MsalServiceException
+    {
+        /// <summary>
+        /// Initializes a new instance of the exception class with a specified
+        /// error code, error message and inner exception indicating the root cause.
+        /// </summary>
+        /// <param name="errorCode">
+        /// The error code returned by the service or generated by the client. This is the code you can rely on
+        /// for exception handling.
+        /// </param>
+        /// <param name="errorMessage">The error message that explains the reason for the exception.</param>
+        /// <param name="source">Specifies the managed identity source from which the exception initiates.</param>
+        public MsalManagedIdentityException(string errorCode, string errorMessage, ManagedIdentitySourceType source) :
+            this(errorCode, errorMessage, null, source)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the exception class with a specified
+        /// error code, error message and managed identity source.
+        /// </summary>
+        /// <param name="errorCode">
+        /// The error code returned by the service or generated by the client. This is the code you can rely on
+        /// for exception handling.
+        /// </param>
+        /// <param name="errorMessage">The error message that explains the reason for the exception.</param>
+        /// <param name="innerException">Represents the root cause of the exception.</param>
+        /// <param name="source">Specifies the managed identity source from which the exception initiates.</param>
+        public MsalManagedIdentityException(string errorCode, string errorMessage, Exception innerException, ManagedIdentitySourceType source) :
+            base(errorCode, errorMessage, innerException)
+        {
+            ManagedIdentitySource = source;
+        }
+
+        /// <summary>
+        /// Specifies the managed identity source from which the exception initiates.
+        /// </summary>
+        public ManagedIdentitySourceType ManagedIdentitySource { get; private set; }
```

Comment:
No need for `private set` - if the modifier is not present, .net will assume the property is read-only i.e. it can only be initialized from the ctor